pkg/virt-controller/watch/vm_test.go

```diff
@@ -4880,6 +4881,146 @@ var _ = Describe("VirtualMachine", func() {
 			Entry("not add default input device by default", nil, []v1.Input{}, nil),
 			Entry("not add default input device when devices already present in VirtualMachine", pointer.Bool(true), []v1.Input{{Name: "existing-0"}}, &v1.Input{Name: "existing-0"}),
 		)
+
+		Context("Live update features", func() {
+			It("should honour the maximum CPU sockets from VM spec", func() {
+				const maxSocketsFromSpec uint32 = 24
+				vm, _ := DefaultVirtualMachine(true)
+				vm.Spec.LiveUpdateFeatures = &virtv1.LiveUpdateFeatures{
+					CPU: &virtv1.LiveUpdateCPU{
+						MaxSockets: kvpointer.P(maxSocketsFromSpec),
+					},
+				}
+
+				vmi := controller.setupVMIFromVM(vm)
+				Expect(vmi.Spec.Domain.CPU.MaxSockets).To(Equal(maxSocketsFromSpec))
+			})
+
+			It("should prefer maximum CPU sockets from VM spec rather than from cluster config", func() {
+				const maxSocketsFromSpec uint32 = 24
+				const maxSocketsFromConfig uint32 = 48
```

Comment:
nit: you can move these under the `Context()` to avoid duplication between all `It`s